jupyterlab_gcedetails/src/components/hardware_scaling_form.spec.tsx

```diff
@@ -0,0 +1,458 @@
+      ...INITIAL_STATE,
+      newConfigurationPrice: NEW_CONFIGURATION_PRICE,
+      configuration: {
+        ...INITIAL_STATE.configuration,
+        attachGpu: true,
+        gpuType: 'NVIDIA_TESLA_K80',
+        gpuCount: '1',
+      },
+    });
+
+    // Select GPU-incompatible machine type
+    const machineTypeSelect = hardwareScalingForm.find(
+      '[label="Machine type"]'
+    );
+    machineTypeSelect.simulate('change', {
+      value: 'incompatible-machine-type',
+      text: 'Incompatible Machine Type',
+    });
+
+    await newConfigurationPrice;
+    hardwareScalingForm.update();
+
+    expect(hardwareScalingForm.state()).toEqual({
+      ...INITIAL_STATE,
+      newConfigurationPrice: NEW_CONFIGURATION_PRICE,
+      configuration: {
+        machineType: {
+          description: 'Incompatible Machine Type',
+          name: 'incompatible-machine-type',
+        },
+        attachGpu: false,
+        gpuType: NO_ACCELERATOR_TYPE,
+        gpuCount: NO_ACCELERATOR_COUNT,
+      },
+    });
+    expect(
+      hardwareScalingForm.find('[label="Attach GPUs"]').prop('disabled')
+    ).toEqual(true);
+  });
+
+  it('Attaches GPU of type ACCELERATOR_TYPE_UNSPECIFIED and count ACCELERATOR_COUNT_UNSPECIFIED if user removes gpu', async () => {
+    const newConfigurationPrice = Promise.resolve(NEW_CONFIGURATION_PRICE);
+    mockGetPrice.mockReturnValue(newConfigurationPrice);
+    const detailsWithGpu = {
+      ...DETAILS,
+      gpu: {
+        cuda_version: '10.1',
+        driver_version: '418.87.01',
+        gpu: 100,
+        memory: 60,
+        name: 'Tesla K80',
+        temperature: 0,
+        count: '1',
+      },
+    };
+    const hardwareScalingForm = shallow(
+      <HardwareScalingForm
+        onSubmit={mockOnSubmit}
+        onDialogClose={mockOnDialogClose}
+        details={detailsWithGpu}
+        priceService={mockPriceService}
+      />
+    );
+    expect(hardwareScalingForm.state()).toEqual({
+      ...INITIAL_STATE,
+      configuration: {
+        ...INITIAL_STATE.configuration,
+        attachGpu: true,
+        gpuType: 'NVIDIA_TESLA_K80',
+        gpuCount: '1',
+      },
+    });
+
+    // Uncheck attach GPU checkbox to remove GPU
+    const checkboxInput = hardwareScalingForm.find('[label="Attach GPUs"]');
+    checkboxInput.simulate('change', {
+      target: { name: 'attachGpu', checked: false },
+    });
+
+    await newConfigurationPrice;
+    hardwareScalingForm.update();
+
+    expect(hardwareScalingForm.state()).toEqual({
+      ...INITIAL_STATE,
+      newConfigurationPrice: NEW_CONFIGURATION_PRICE,
+    });
+
+    // Submit form
+    hardwareScalingForm.find(ActionBar).prop('onPrimaryClick')();
+
+    expect(mockOnSubmit).toHaveBeenCalledWith({
+      ...INITIAL_STATE.configuration,
+      attachGpu: true,
+      gpuType: NO_ACCELERATOR_TYPE,
+      gpuCount: NO_ACCELERATOR_COUNT,
+    });
+  });
+
+  it('Displays correct pricing information', async () => {
```

Comment:
I believe you could use snapshot tests here rather than the string matching if you'd like.